addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+/**
+ * Declare a MapType where key is a string and value is an array of type string
+ * Use for storing a file's path as key and its dependent module's relative path
+ */
+declare type ImportMap {
+  [key: string]: ImportMap | string[];
+}
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createtsSourceFile(fileName: string): ts.SourceFile {
+  return ts.createSourceFile(fileName,
+                             fs.readFileSync(fileName).toString(),
+                             ts.ScriptTarget.ES6,
+                             true);
+}
+
+/**
+ *  Store all the files' path under the given path in a passed in array.
+ *  All the files in the subdirectories of the given path are also included.
+ * 
+ * @param filePath path from where all the files' path are retrieved
+ * @param allFiles array where the files are stored and then returned.
+ * 
+ */
+
+export function getFiles(filePath: string, allFiles: string[]): any {
+  let files = fs.readdirSync(filePath);
+  files.forEach(file => {
+    if (path.extname(file) === '') {
+      getFiles(path.join(filePath, file), allFiles);
+    } else {
+      allFiles.push(path.join(filePath, file));
+    }
+  });
+  return allFiles;
+}
+
+/**
+ * Traverse through ts.Node recursively and stores moduleSpecifiers in an array
+ * 
+ * @param {ts.Node} node: Typescript Node of whose AST is being traversed
+ * @param {string[]} moduleSpecifiers: array to store moduleSpecifiers if found.
+ * 
+ */
+export function getModuleSpecifier(node: ts.Node, moduleSpecifiers: string[]): any {
+  node.getChildren().forEach(nodeObj => {
```

Comment:
Here's a cool trick; you can use reduce to make your code more flow-like:

```typescript
declare interface ModuleImport {
  specifier: string,
  pos: number,
  end: number
};

function _getImportClauses(node: ts.Node): ModuleImport[] {
  return node.getChildren()
    .reduce(function(nodes, node) {
      if (current.kind === ts.SyntaxKind.ImportClause) {
        return nodes.concat({ specifier: node.text, pos: node.pos, end: node.end });
      } else {
        return nodes.concat(_getImportClauses(node));
      }
    }, [])
}
```

You don't have to if you don't want to do it, it just reads a bit better but your version isn't bad :)